app/controllers/records.js

```diff
@@ -51,22 +61,9 @@ export default Controller.extend({
     }));
   }),
 
-  filtered: computed('searchValue', 'model.@each.{columnValues,filterValues}', 'filterValue', function() {
-    let search = this.searchValue;
-    let filter = this.filterValue;
-
-    return this.model.filter(item => {
-      // check filters
-      if (filter && !get(item, `filterValues.${filter}`)) {
-        return false;
-      }
-
-      // check search
-      if (!isEmpty(search)) {
-        let searchString = this.recordToString(item);
-        return !!searchString.match(new RegExp(`.*${escapeRegExp(search.toLowerCase())}.*`));
-      }
-      return true;
+  filteredRecords: computed('searchValue', 'model.@each.{columnValues,filterValues}', 'filterValue', function() {
+    return this.model.filter((record) => {
+      return this.passesFilter(record) && this.passesSearch(record);
```

Comment:
The logic for `passesFilter` and `passesSearch` seems fairly simple. Would you be okay with not pulling out into separate functions and leaving inside the computed?